### PR TITLE
f/SRV-2738: Cucumber Xray Jira Integration

### PR DESCRIPTION
**Pull request by patrick-sf**
This PR: 
- adds `publish-unit-test-results-via-xray` and `publish-integration-test-results-via-xray` commands
- adds `publish-unit-test-results-to-xray` and `publish-integration-test-results-to-xray` jobs
- persists mocha and cucumber unit + integration test reports to workspace to enable usage across jobs
- removes `push-test-results-xray.sh` script as the logic has been transferred

### CODE REVIEW

**Review by misiek-sf — APPROVED** [review]

**Review by codin-mgm — APPROVED** [review]

File: microservices.yml [hunk 1]
```diff
@@ -451,37 +461,127 @@ commands:
...
+              echo "cucumber test results file not found ($TEST_RESULTS_FILE)"
+              exit 1
+            fi
+            XRAY_AUTH_TOKEN=$(\
+              curl -s https://xray.cloud.xpand-it.com/api/v1/authenticate \
+                -H "Content-Type: application/json" -X POST \
+                --data "{ \"client_id\": \"$XRAY_CLIENT_ID\", \"client_secret\": \"$XRAY_CLIENT_SECRET\" }" \
+                | tr -d '"'
+            )
+            RESULTS_TEST=$(\curl "https://xray.cloud.xpand-it.com/api/v1/import/execution/cucumber" \
```
Comment (afaq-sf) [inline microservices.yml:517]: Is there any need to assign this command's output to a variable?

Seems like the only thing we do with it is `echo` it to stdout. 
I believe we can just skip that by just not assigning output to the variable

File: microservices.yml [hunk 2]
```diff
@@ -451,37 +461,127 @@ commands:
...
+          name: Upload Mocha unit test results to Jira Xray
           when: always
           command: |
-            # get push-test-results-to-xray.sh
-             wget https://raw.githubusercontent.com/MGMDV-Orbs/microservices/f/xray-r2.3/.circleci/push-test-results-to-xray.sh -O push-test-results-to-xray.sh
-      - run:
-          name: Upload Mocha unit test results to Jira Xray
+            TEST_RESULTS_FILE=./reports/unit-test-mocha-results.xml
+            PROJECT_CODE=SRV
+            TEST_TYPE=junit
```
Comment (afaq-sf) [inline microservices.yml:475]: Seems like this variable is only used in one place and that is to echo to stdout for informational purposes.
does not seem like it is used for any flow control or re-use

Probably a good idea to not initialize unnecessary variables

File: microservices.yml (hunk 1 above)
Comment (afaq-sf) [reply microservices.yml:517]: Also, do we know if a failing call from the subcommand will fail the circleci step properly?

**Review by afaq-sf — APPROVED** [review]

**Comment by afaq-sf** [comment]
I see a lot of commits that seem to be just discovery-phase iterations for syntax and flow.

In the future, it might be helpful to work that out in a POC branch and keep the feature branch to the meaningful set of digestible behavior changes.

File: microservices.yml (hunk 1 above)
Comment (patrick-sf) [reply microservices.yml:517]: The check for the file will fail the circleci step. A failed post to xray to integrate the results will not fail the circleci step.

**Pull request closed without merging**